repo/mk/core.deps.git-hook-pre-push.mk

```diff
@@ -0,0 +1,19 @@
+# install a pre-push git hook that runs 'make check' on 'git push', before pushing to remote
+
+# the hook can be ignored one time by running 'git push --no-verify'
+
+# the hook can be not-installed by adding to the Makefile:
+# SF_DEPS_TARGETS := $(filter-out .git/hooks/pre-push,$(SF_DEPS_TARGETS))
+
+ifneq ($(wildcard .git),)
+SF_DEPS_TARGETS := \
+	$(SF_DEPS_TARGETS) \
+	.git/hooks/pre-push \
+
+endif
+
+# ------------------------------------------------------------------------------
+
+.git/hooks/pre-push:
+	$(MKDIR) $(shell dirname $@)
+	$(CP) $(SUPPORT_FIRECLOUD_DIR)/repo/dot.git/hooks/pre-push $@
```

Comment:
Awesome! I don't understand how this works, though. I see that it appends to some "SF_DEPS_TARGETS" which I assume is part of the "deps" target? Then I don't understand what the "$@" in the target is. mkdir? In the end, this hook need to be put in `.git/hooks/pre-push`? That path is in $@?

Reply:
https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html

`$@ = The file name of the target of the rule.` in this case a synonym to `.git/hooks/pre-push`

Reply:
Ah, that clears it all up! Arigato gozaimashita.